File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
/*
Motivation: Buy and sell every day, if it worth it.
    For every day, if the price is higher than yesterday, then we could have
    bought it and sell today. other wise we won't buy.
Complexity: 
    Time - O(n): Looping over the array once.
    Space - O(1): Auxiliary.
*/
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;

        // calculate profit from each day
        for (int i = 1; i < prices.size(); ++i) {
            // yesterday's price is lower than today's
            if (prices[i - 1] < prices[i]) {
                profit += prices[i] - prices[i - 1];
            }
        }

        return profit;
    }
};

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        # Can't profit from one day or less
        if len(prices) < 1:
            return 0

        profit = 0

        # calculate profit from each day
        for i in range(1, len(prices)):

            # today's price is lower than yesterday's
            if prices[i - 1] < prices[i]:
                profit += prices[i] - prices[i - 1]

        return profit
            
            


        